api/src/infrastructure/docker.rs

```diff
@@ -128,130 +142,149 @@ impl DockerInfrastructure {
+            .and_then(|n| n.id);
 
         if let Some(n) = network_id {
             return Ok(n);
         }
 
-        debug!("Creating network for app {}.", app_name);
+        debug!("Creating network for app {app_name}.");
 
         let network_create_info = docker
-            .networks()
-            .create(&NetworkCreateOptions::builder(network_name.as_ref()).build())
+            .create_network(CreateNetworkOptions::<&str> {
+                name: &network_name,
+                ..Default::default()
+            })
             .await?;
+        let network_id = network_create_info
+            .id
+            .expect("id is mandatory for a Docker Network.");
 
-        debug!(
-            "Created network for app {} with id {}",
-            app_name, network_create_info.id
-        );
+        debug!("Created network for app {app_name} with id {network_id}");
 
-        Ok(network_create_info.id)
+        Ok(network_id)
     }
 
-    async fn connect_traefik(&self, network_id: &String) -> Result<(), ShipLiftError> {
-        let docker = Docker::new();
+    async fn connect_traefik(&self, network_id: &str) -> Result<(), BollardError> {
+        let docker = Docker::connect_with_socket_defaults()?;
 
         let containers = docker
-            .containers()
-            .list(&ContainerListOptions::builder().build())
+            .list_containers(Some(ListContainersOptions::<&str> {
+                all: true,
+                ..Default::default()
+            }))
             .await?;
+
         let traefik_container_id = containers
             .into_iter()
-            .find(|c| c.image.contains("traefik"))
-            .map(|c| c.id);
+            .find(|c| c.image.as_ref().map_or(false, |s| s.contains("traefik")))
+            .and_then(|c| c.id);
 
         if let Some(id) = traefik_container_id {
             if let Err(e) = docker
-                .networks()
-                .get(network_id)
-                .connect(&ContainerConnectionOptions::builder(&id).build())
+                .connect_network(
+                    network_id,
+                    ConnectNetworkOptions::<&str> {
+                        container: &id,
+                        ..Default::default()
+                    },
+                )
                 .await
             {
-                debug!("Cannot traefik: {}", e);
+                debug!("Cannot traefik: {e}");
             }
         }
 
         Ok(())
     }
 
-    async fn disconnect_traefik(&self, network_id: &String) -> Result<(), ShipLiftError> {
-        let docker = Docker::new();
-
+    async fn disconnect_traefik(&self, network_id: &str) -> Result<(), BollardError> {
+        let docker = Docker::connect_with_socket_defaults()?;
         let containers = docker
-            .containers()
-            .list(&ContainerListOptions::builder().build())
+            .list_containers(Some(ListContainersOptions::<&str> {
```

Comment:
during connecting the containers there is usage of `all` options. Shouldn't we use that here as well?

Reply:
Agreed. In shiplift too the default build option is `all`.

Reply:
There should have been no `all` at both sides. Now its in-line with prior shiplift implementation.
```rust
async fn connect_traefik(&self, network_id: &String) -> Result<(), ShipLiftError> {
        let docker = Docker::new();

        let containers = docker
            .containers()
            .list(&ContainerListOptions::builder().build())
            .await?;
```